### PR TITLE
Modify PIDNet conv bias, add head_list property on models

### PR DESCRIPTION
**Pull request by illian01**
## Description

Please include a summary in English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #(issue)

You should link at least one existing issue for PR. Before your create a PR, please check to see if there is an issue for this change.  
PRs from forked repository not accepted.

## Change(s)

- Modify PIDNet conv1 bias as `True` according to [original repo](https://github.com/XuJiacong/PIDNet/blob/fefa51716bddc13a4321af2c70a074367100645a/models/pidnet.py#L24-L31)
- Add `head_list` property on models to regardlessly handle `TaskModel` or `full-models`.

## Changelog

If you PR to `dev` branch, please add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file and include a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`.

### CODE REVIEW

File: src/netspresso_trainer/models/full/experimental/pidnet.py [hunk 1]
```diff
@@ -38,10 +38,10 @@ def __init__(
         self.conv1 = nn.Sequential(
             ConvLayer(in_channels=3, out_channels=planes,
                       kernel_size=3, stride=2, padding=1,
-                      norm_type='batch_norm', act_type='relu'),
+                      norm_type='batch_norm', act_type='relu', bias=True),
```
Comment (shinkookchoi) [inline src/netspresso_trainer/models/full/experimental/pidnet.py:41]: The custom convolution layer has 'bias=False' set as the default value. 
However, in '[torch.nn.Conv2d](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html)', 'bias=True' is the default value. 

Could you please clarify the reason behind setting 'bias=False' as the default value in the custom layer?

Comment (illian01) [reply src/netspresso_trainer/models/full/experimental/pidnet.py:41]: Our custom `ConvLayer` has conv-bn-act pattern.
And, convolution before batch normalization usually has no bias because batch normalization takes a role same with convolution bias, thus it cancels effect of the bias.
Please refer [Batch normalization paper](https://arxiv.org/pdf/1502.03167.pdf) section 3.2 second paragraph.

Comment (shinkookchoi) [reply src/netspresso_trainer/models/full/experimental/pidnet.py:41]: Thank you for the clarification.

**Review by shinkookchoi — APPROVED** [review]
